### PR TITLE
Add a .cnb extension by default on buildpackage creation

### LINKED ISSUE #612: Append .cnb to package files created without file extensions
### Description
Append `.cnb` to package files created using `pack package-buildpack <NAME> -f file`, when there is no extension provided.  

Otherwise, it is confusing on two respects: (1) users won't know what a `.cnb` file, and (2) users may think it's a `zip` file, when it's encoded differently. 

### Additional context
Discussed [here](https://buildpacks.slack.com/archives/CJ6B92ZSB/p1586961828112700?thread_ts=1586959710.111700&cid=CJ6B92ZSB) on the Buildpacks slack
Mentioned https://github.com/buildpacks/pack/issues/587#issuecomment-625468400 as well.

### PR DESCRIPTION
**Pull request by samj1912**
Signed-off-by: Sambhav Kothari <skothari44@bloomberg.net>

## Summary

This change adds a .cnb extension by default when the output format is a file.

Fixes #612 

#### Before

```bash
$ pack buildpack package test -f file                                                                                                                       

Successfully created package test

$ pack buildpack package test.cnb -f file                                                                                                                   

Successfully created package test.cnb

$ pack buildpack package test.tar.gz -f file                                                                                                                Successfully created package test.tar.gz
```

#### After

```bash
$ pack-dev buildpack package test -f file                                                                                                                   

Successfully created package test.cnb

$ pack buildpack package test.cnb -f file                                                                                                               

Successfully created package test.cnb

$ pack buildpack package test.xz -f file                                                                                                                

Warning: .xz is not a valid extension for a Buildpackage. Buildpackages must have a .cnb extension
Successfully created package test.xz
```

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [ ] No

## Related

Resolves #612

### CODE REVIEW

File: internal/commands/buildpack_package.go [hunk 1]
```diff
@@ -76,6 +76,15 @@ func BuildpackPackage(logger logging.Logger, cfg config.Config, client Buildpack
 			}
 
 			name := args[0]
+			if flags.Format == pack.FormatFile {
+				switch ext := filepath.Ext(name); ext {
+				case pack.CNBExtension:
+				case "":
+					name += pack.CNBExtension
+				default:
+					logger.Warnf("%s is not a valid extension for a Buildpackage. Buildpackages must have a %s extension", style.Symbol(ext), style.Symbol(pack.CNBExtension))
```
Comment (dfreilich) [inline internal/commands/buildpack_package.go:85]: We don't use the term `buildpackages` in the Long description, so we probably should standardize what we are calling it – either a `packaged buildpack`, or a `buildpackage`.

**Review by dfreilich — CHANGES_REQUESTED** [review]
Can you change the example of the command to use a `.cnb` package name? 

We may also want to clarify in our Long description that `.cnb` is our file extension for OCI packaged buildpacks (to make it a bit clearer for newer contributors/users what a `.cnb` file is)

File: internal/commands/buildpack_package.go (hunk 1 above)
Comment (samj1912) [reply internal/commands/buildpack_package.go]: Renamed to packaged buildpack.

**Comment by samj1912** [comment]
> Can you change the example of the command to use a `.cnb` package name?
> 
> We may also want to clarify in our Long description that `.cnb` is our file extension for OCI packaged buildpacks (to make it a bit clearer for newer contributors/users what a `.cnb` file is)

Done 💯

**Review by dwillist — APPROVED** [review]
Simple acceptance  ✅

**Comment by samj1912** [comment]
Should this be good to go in now that 0.18 is out?

**Review by dfreilich — APPROVED** [review]

**Pull request merged**